shell/client/grainview.js

```diff
@@ -0,0 +1,482 @@
+var counter = 0;
+
+GrainView = function GrainView(grainId, path, query, hash, token, parentElement) {
+  // Owned grains:
+  // grainId, path, query, hash, dep.
+  //   callback sets error, openingSession on failure,
+  //                 grainId, sessionId, sessionSub on success.
+  //
+  // Sturdyref ApiTokens:
+  // grainId, path, query, hash, dep.
+  //   callback sets error, openingSession on failure
+  //                 grainId, sessionId, sessionSub on success.
+  //
+  // Token-only sessions:
+  // grainId, token, path, query, hash, dep
+  //   callback sets error, openingSession on failure
+  //                 grainId, sessionId, title, and session Sub on success
+
+  this._grainId = grainId;
+  this._originalPath = path;
+  this._originalQuery = query;
+  this._originalHash = hash;
+  this._token = token;
+
+  this._status = "closed";
+  this._revealIdentity = undefined; // set to true or false to make explicit
+  this._dep = new Tracker.Dependency();
```

Comment:
If you made this object's mutable members instances of `ReactiveVar`, would that eliminate the need for this manual Dependency? Would that make things simpler?